test/ex_machina/ecto_test.exs

```diff
@@ -2,17 +2,50 @@ defmodule ExMachina.EctoTest do
   use ExUnit.Case, async: true
 
   defmodule TestRepo do
-    def insert!(record) do
-      send self, {:created, record}
-      record
+    def insert!(model) do
+      send self, {:created, model}
+      # simulate saving of model by adding id
+      model |> Map.put(:id, 1)
```

Comment:
If you send the model with the id on it already that could clean up the assertions and make them a bit more robust

```
model = %{model | id: 1}
send self, {:create, model}
model
```